src/components/templates/AnimateMap/AnimateMap.tsx

```diff
@@ -0,0 +1,85 @@
+import * as PIXI from "pixi.js";
+
+import React, { useState, useRef, useEffect } from "react";
+import { Provider, ReactReduxContext } from "react-redux";
+import { AppConsumer, Container, Stage, useApp } from "@inlet/react-pixi";
+import { AnimateMapVenue } from "../../../types/venues";
+import { useUser } from "hooks/useUser";
+import { useSelector } from "../../../hooks/useSelector";
+import { animateMapStageOptionsSelector } from "../../../utils/selectors";
+
+import "./AnimateMap.scss";
+
+import { MapContainer } from "./components/Map/MapContainer";
+import { MAP_IMAGE } from "./constants/Resources";
+
+export interface AnimateMapProps {
+  venue: AnimateMapVenue;
+}
+
+export const AnimateMap: React.FC<AnimateMapProps> = () => {
+  const options = useSelector(animateMapStageOptionsSelector);
+
+  const containerRef = useRef<HTMLDivElement>(null);
+  const { user, profile } = useUser();
+
+  if (!user || !profile) {
+    return <>Loading..</>;
+  }
+
+  const container = containerRef.current ?? null;
+
+  return (
+    <div ref={containerRef} className="animatemap-venue-container">
+      {container && (
+        <ReactReduxContext.Consumer>
```

Comment:
@0xdevalias I consulted with Vasily about his decision. We are facing a known issue with react-pixi. We only use this wrapper to patch the provider. And then we use only hooks according to your recommendations.
We can encapsulate the wrapper in a separate component, but we don't see a way to get rid of it entirely.
If you have an idea how to fix this, I will be glad to consider it.
PS https://github.com/inlet/react-pixi/issues/77

Reply:
Makes sense, thanks for the context. Following some of those links I ended up at this section of the README:

- https://github.com/michalochman/react-pixi-fiber#react-context-api-limitations

This links to the following React issue:

- https://github.com/facebook/react/issues/13336

Which seems to have been closed by this PR:

- https://github.com/facebook/react/pull/13728

Which seems to have been included in the React `16.6.0` changelog in this PR:

- https://github.com/facebook/react/pull/13927

Since we seem to be using `16.13.1`, my guess/theory is that we should have that fix available to us:

- https://github.com/sparkletown/sparkle/blob/staging/package.json#L56

I'm not really sure of the specifics of how this issue works, how it was fixed, or how that applies to the lib you're using here, but it would be good if you could quickly see if it is resolved in a 'proper' way now, and if so, align the code here to that 'proper' way.

If it's going to take too long/be too hard/etc, then can you add the following comment (replacing the `XXX`'s with the relevant details) just before the `return` JSX so we have a note captured for future:

```
// @debt We need to use Redux's XXX here to work around a limitation in react-pixi-fibre as per:
//   https://github.com/michalochman/react-pixi-fiber#react-context-api-limitations
//
// While this issue was resolved in React in v16.6.0 via facebook/react#13728, we can't use that
//   fix here yet because XXX. Once that is resolved, we should refactor this accordingly.
```

Reply:
Also, if we still do need to hack around it like this, I wonder if we can still simplify/move the code here to a 'closer to Redux hooks' based implementation by making use of `useStore` instead of `ReactReduxContext.Consumer`, etc:

- https://react-redux.js.org/api/hooks#usestore
- https://react-redux.js.org/using-react-redux/accessing-store